src/Features/Core/Portable/FindUsages/SourceReferenceItem.cs

```diff
@@ -35,48 +36,65 @@ internal sealed class SourceReferenceItem
         public bool IsWrittenTo { get; }
 
         /// <summary>
-        /// Additional information about the reference.
+        /// Additional properties for the reference that can have multiple values.
         /// Each entry represents a key-values pair of data. For example, consider the below entry:
         ///     { "ValueUsageInfo" } = { "Read", "Write" }
         /// This entry indicates that the reference has additional value usage information which indicate
         /// it is a read/write reference, such as say 'a++'.
         /// </summary>
-        public ReferenceInfoMap ReferenceInfo { get; }
+        public AdditionalPropertiesWithMultipleValuesMap AdditionalPropertiesWithMultipleValues { get; }
+
+        /// <summary>
+        /// Additional properties for the reference that can have only one value.
+        /// For example, { "ContainingTypeInfo" } = { "MyClass" }
+        /// </summary>
+        public ImmutableArray<FindUsageProperty> FindUsagesProperties { get; }
```

Comment:
feels strange that we can't unify this wiht the above map...

Reply:
why do we need two separate ways to add additional information to the reference?  why can't we have one unified way to just add more columns?  then we can have columns for containing type/member, and a column for the info about the reference.